x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/DataFrameAnalyticsTask.java

```diff
@@ -161,7 +162,11 @@ private void cancelReindexingTask(String reason, TimeValue timeout) {
         cancelReindex.setTaskId(reindexTaskId);
         cancelReindex.setReason(reason);
         cancelReindex.setTimeout(timeout);
-        CancelTasksResponse cancelReindexResponse = client.admin().cluster().cancelTasks(cancelReindex).actionGet();
+
+        // We need to cancel the reindexing task within context with ML origin as we started the task
+        // from the same context
+        CancelTasksResponse cancelReindexResponse = cancelTaskWithinMlOriginContext(cancelReindex);
```

Comment:
I guess I need to learn more about context/origins etc.
What are the rules one should follow when spawning requests from ML code?
I've seen at least two ways of setting `ML_ORIGIN`, namely `threadContext.stashWithOrigin(ML_ORIGIN)` and using `OriginSettingClient`.
Also, it is possible not to set the origin (like this code before your fix). Could we make it so that all ML calls set the origin?

Reply:
`stashWithOrigin` and `OriginSettingClient` both do the same thing.

> Could we make it so that all ML calls set the origin?

No, some calls we make need to run with the credentials of the user who called an endpoint or the user who created or updated a job.  We need to make sure that ML isn't a means to gain access to data you shouldn't have access to.  But the internal housekeeping ML does execute with `ML_ORIGIN`.

Reply:
Ok, thanks!